mico-core/src/main/java/io/github/ust/mico/core/web/ServiceController.java

```diff
@@ -359,15 +361,22 @@
     }
 
    @GetMapping("/{" + PATH_VARIABLE_SHORT_NAME + "}/{" + PATH_VARIABLE_VERSION + "}" + "/dependencyGraph")
-    public ResponseEntity<Resources<Resource<MicoService>>> getDependencyGraph(@PathVariable(PATH_VARIABLE_SHORT_NAME) String shortName,
+    public ResponseEntity<Resource<MicoServiceDependencyGraphDTO>> getDependencyGraph(@PathVariable(PATH_VARIABLE_SHORT_NAME) String shortName,
                                                                                @PathVariable(PATH_VARIABLE_VERSION) String version) {
-        MicoService micoService = getServiceFromDatabase(shortName, version);
-        List<MicoService> micoServices = serviceRepository.getAllDependeesOfMicoService(micoService.getShortName(), micoService.getVersion());
-        List<Resource<MicoService>> resourceList = getServiceResourcesList(micoServices);
-        return ResponseEntity.ok(
-            new Resources<>(resourceList,
-                linkTo(methodOn(ServiceController.class).getDependencyGraph(shortName, version)).withSelfRel()));
-    }
+       MicoService micoServiceRoot = getServiceFromDatabase(shortName, version);
+       List<MicoService> micoServices = serviceRepository.getAllDependeesOfMicoService(micoServiceRoot.getShortName(), micoServiceRoot.getVersion());
+       MicoServiceDependencyGraphDTO micoServiceDependencyGraph = new MicoServiceDependencyGraphDTO().setMicoServices(micoServices);
+       LinkedList<MicoServiceDependencyGraphEdgeDTO> micoServiceDependencyGraphEdgeList = new LinkedList<>();
+       for (MicoService micoService : micoServices) {
+           micoService.getDependencies().forEach(micoServiceDependency -> {
```

Comment:
dependencies may not be initialized correctly by ogm because the services were loaded with a custom query.
In fact it fails with the service configuration of the insertTestValues script in master for the service `short-name-david` with version `4.2.3` if a dependency from `short-name-dude` version `0.5.6` to `short-name-dominik` version `5.6.1` is added.